pjmedia/src/pjmedia/vid_codec.c

```diff
@@ -200,6 +201,18 @@ PJ_DEF(pj_status_t) pjmedia_vid_codec_mgr_register_factory(
     if (status != PJ_SUCCESS)
         return status;
 
+    if ((factory->op->test_alloc)(factory, info) == PJ_SUCCESS) {
+        status = (factory->op->default_attr)(factory, info, &param);
+        if (status == PJ_SUCCESS) {
+            if ((param.enc_fmt.det.vid.size.w % 2 == 1) ||
+                (param.enc_fmt.det.vid.size.h % 2 == 1))
+            {
+                PJ_LOG(3, (THIS_FILE, "Video resolution must be even"));
+                return PJ_EINVAL;
+            }
+        }
+    }
+
```

Comment:
`info` is an array of codecs, so I think this block should be moved to inside the codec loop below?

Reply:
IMO either approach is fine as long as the video processing components (including third-party converters/codecs) get even resolution.